src/ValidatorKeys.cpp

```diff
@@ -46,9 +46,9 @@ ValidatorKeys::ValidatorKeys (KeyType const& keyType)
     : keyType_ (keyType)
     , tokenSequence_ (0)
     , revoked_ (false)
+    , publicKey_ (generateKeyPair (keyType_, randomSeed ()).first)
+    , secretKey_ (generateKeyPair (keyType_, randomSeed ()).second)
```

Comment:
This is going to give you two unrelated keys. A better way to ensure you're working with the same keypair is to add a `std::pair` member, initialize that first, then initialize pk and sk from that.
In the .h
```
std::pair<PublicKey, SecretKey> const keyPair_;
PublicKey const& publicKey_;
SecretKey const& secretKey_;
```
In the ctor
```
, keyPair_ (generateKeyPair (keyType_, randomSeed ()))
, publicKey_ (keyPair_.first)
, secretKey_ (keyPair_.second)
```

Reply:
I have done something slightly different. I feel my solution has lesser repetition of information. But I'm doing this blind, I've been able to run a local CI test ://